### PR TITLE
Add documentation guide of the CustomerSync.

### PR DESCRIPTION
**Pull request by ninalindl**
#### Summary

User guide and release notes draft for customer sync were added to the docs.
#### Description

The structure for the user guide was adapted from the Cart Discount Sync user guide. 
Unique properties of customer sync, like unique Address keys, references to customer groups and custom types are mentioned.
Links to tests as examples for the use of customer sync, as well as links to the javadoc need to be added once customer sync is merged into master.
#### Todo

- Tests
    - [ ] Unit 
    - [ ] Integration
- [x] Documentation
- [x] Add Release Notes entry.

#### Hints for Review

Check RELEASE_NOTES.md and CUSTOMER_SYNC.md

### CODE REVIEW

File: docs/usage/CUSTOMER_SYNC.md [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+__Note__ The statistics object contains the processing time of the last batch only. This is due to two reasons:
+
+ 1. The sync processing time should not take into account the time between supplying batches to the sync.
+ 2. It is not known by the sync which batch is going to be the last one supplied.
+
+#### More examples of how to use the sync
+
+ [Sync from an external source](https://github.com/commercetools/commercetools-sync-java/tree/master/src/integration-test/java/com/commercetools/sync/integration/externalsource/customers/CustomerSyncIT.java).
+
+*Make sure to read the [Important Usage Tips](IMPORTANT_USAGE_TIPS.md) for optimal performance.*
```
Comment (ahmetoz) [inline docs/usage/CUSTOMER_SYNC.md:85]: Those are changed with the latest master, so you need to show example usages of the sync options.

Comment (ahmetoz) [reply docs/usage/CUSTOMER_SYNC.md:85]: For example https://github.com/commercetools/commercetools-sync-java/blob/master/docs/usage/CART_DISCOUNT_SYNC.md#about-syncoptions

File: docs/usage/CUSTOMER_SYNC.md [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+````java
+Optional<UpdateAction<Customer>> updateAction = CustomerUpdateActionUtils.buildChangeEmailAction(oldCustomer, customerDraft);
+````
+
+More examples for particular update actions can be found in the test scenarios for [CustomerUpdateActionUtils](https://github.com/commercetools/commercetools-sync-java/tree/master/src/test/java/com/commercetools/sync/customers/utils/CustomerUpdateActionUtilsTest.java)
+and [AddressUpdateActionUtils](https://github.com/commercetools/commercetools-sync-java/tree/master/src/test/java/com/commercetools/sync/customers/utils/AddressUpdateActionUtilsTest.java).
+
+
+##Caveats
+Customer passwords are not being synced.
```
Comment (ahmetoz) [inline docs/usage/CUSTOMER_SYNC.md:105]: Here this is only for update cases, not for the creation of a new customer, so can you describe it in more clear way.

File: docs/usage/CUSTOMER_SYNC.md (hunk 1 above)
Comment (ninalindl) [reply docs/usage/CUSTOMER_SYNC.md:180]: I added the part about SyncOptions: d3048593.

File: docs/usage/CUSTOMER_SYNC.md (hunk 2 above)
Comment (ninalindl) [reply docs/usage/CUSTOMER_SYNC.md:200]: What happens if a new customer is created? It is created with all the information that could be synced but without a password?

Comment (ninalindl) [reply docs/usage/CUSTOMER_SYNC.md]: 78163328

File: docs/usage/CUSTOMER_SYNC.md [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+ final Function<CustomerDraft, CustomerDraft> draftFunction =
+             customerDraft -> CustomerDraftBuilder.of(customerDraft)
+                                                  .key(customerDraft.getKey() + "_filteredKey")
+                                                  .build();
+ 
+final CustomerSyncOptions customerSyncOptions = CustomerSyncOptionsBuilder
+                                                  .of(CTP_CLIENT)
+                                                  .beforeCreateCallback(draftFunction)
+                                                  .build();
+````
```
Comment (leungkinghin) [inline docs/usage/CUSTOMER_SYNC.md:136]: Here we discussed with Roman a while ago. It is not easy to list out a practical example for beforeCreateCallback. We can follow the approach in other resources documents
https://github.com/commercetools/commercetools-sync-java/blob/master/docs/usage/CATEGORY_SYNC.md#4-beforecreatecallback

File: docs/usage/CUSTOMER_SYNC.md [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+
+##### 5. `batchSize`
+A number that could be used to set the batch size with which customers are fetched and processed,
+as customers are obtained from the target project on commercetools platform in batches for better performance. The 
+algorithm accumulates up to `batchSize` resources from the input list, then fetches the corresponding cart discounts 
+from the target project on commercetools platform in a single request. Playing with this option can slightly improve or 
+reduce processing speed. If it is not set, the default batch size is 50 for customer discount sync.
```
Comment (leungkinghin) [inline docs/usage/CUSTOMER_SYNC.md:143]:
```suggestion
reduce processing speed. If it is not set, the default batch size is 50 for customer sync.
```

File: docs/usage/CUSTOMER_SYNC.md [hunk 5]
```diff
@@ -0,0 +1,201 @@
...
+final CustomerSyncOptions customerSyncOptions = CustomerSyncOptionsBuilder
+                                                  .of(CTP_CLIENT)
+                                                  .beforeCreateCallback(draftFunction)
+                                                  .build();
+````
+
+##### 5. `batchSize`
+A number that could be used to set the batch size with which customers are fetched and processed,
+as customers are obtained from the target project on commercetools platform in batches for better performance. The 
+algorithm accumulates up to `batchSize` resources from the input list, then fetches the corresponding cart discounts
```
Comment (leungkinghin) [inline docs/usage/CUSTOMER_SYNC.md:141]:
```suggestion
algorithm accumulates up to `batchSize` resources from the input list, then fetches the corresponding customers
```

**Review by ahmetoz — APPROVED** [review]

**Review by leungkinghin — CHANGES_REQUESTED** [review]
So we are going to release v2.3.0 after customer sync.
Besides the CUSTOMER_SYNC.md, as i remember, the version number is written in all other resources sync documents. Can you go to have a check if we have to update the version number from v2.2.1 to v2.3.0 in page like PRODUCT_SYNC.md, CATEGORY_SYNC.md, etc.

**Comment by ahmetoz** [comment]
> So we are going to release v2.3.0 after customer sync.
> Besides the CUSTOMER_SYNC.md, as i remember, the version number is written in all other resources sync documents. Can you go to have a check if we have to update the version number from v2.2.1 to v2.3.0 in page like PRODUCT_SYNC.md, CATEGORY_SYNC.md, etc.

You are right but let's handle the version changes in a different PR.

File: docs/usage/CUSTOMER_SYNC.md (hunk 3 above)
Comment (ninalindl) [reply docs/usage/CUSTOMER_SYNC.md:136]: So I just add "Please refer to the [example in the product sync document](https://github.com/commercetools/commercetools-sync-java/blob/master/docs/usage/PRODUCT_SYNC.md#example-set-publish-stage-if-category-references-of-given-product-draft-exists)." like in Category sync?

Comment (leungkinghin) [reply docs/usage/CUSTOMER_SYNC.md:136]: yes :)

**Review by leungkinghin — APPROVED** [review]

File: docs/usage/CUSTOMER_SYNC.md (hunk 3 above)
Comment (ninalindl) [reply docs/usage/CUSTOMER_SYNC.md]: 0f595cbd

**Pull request merged**